internal/database/actions.go

```diff
@@ -17,10 +17,41 @@ import (
 	"github.com/upper/db/v4"
 )
 
+var (
+	errInvalidDatabasePath = "database: passed an empty database path"
+)
+
+// DatabaseConfig is the configuration for creating a new database instance
+type DatabaseConfig struct {
+	DatabasePath string
+}
+
+// NewDatabase returns a new database instance
+func NewDatabase(config *DatabaseConfig) (*Database, error) {
+	if config.DatabasePath == "" {
+		return &Database{}, errors.New(errInvalidDatabasePath)
+	}
+	db, err := Connect(config.DatabasePath)
+	return &Database{
+		sess:   db,
+		dbPath: config.DatabasePath,
+	}, err
```

Comment:
I would rather like see the implementation "implement" the equivalent of Rust's `Result<T>` construct and _either_ return a nil database to the caller (and a non-nil error) _or_ return a valid database (and a nil error).